src/flib/pionfput_mod.F90.in

```diff
@@ -442,12 +457,15 @@ contains
 !
     clen = len(ival)
     sd = size(ival)
-    allocate(cval(clen*sd))
-    call Fstring2Cstring_{DIMS}d (ival, cval)
-
-    ierr = PIOc_put_vara_text(file%fh, varid-1,  cstart, ccount, cval)
-    deallocate(cval, cstart, ccount)
-
+    if (clen*sd > 0) then
+        allocate(cval(clen*sd))
+        call Fstring2Cstring_{DIMS}d (ival, cval)
+
+        ierr = PIOc_put_vara_text(file%fh, varid-1,  cstart, ccount, cval)
+        deallocate(cval, cstart, ccount)
+    else
+        print *, 'PIO: WARNING: Empty {DIMS}d text data passed to PIO_put_var function put_vara_{DIMS}d_text, a collective C API call will be skipped on this processor. ', __PIO_FILE__, __LINE__
```

Comment:
Looks good. Can you change "processor" to "process" in the message (and other messages)?

Reply:
No problem, I can change that. FYI, in E3SM code's comments, there are more occurrences of "on this processor" than "on this process", though.